deep_qa/layers/attention/matrix_attention.py

```diff
@@ -1,10 +1,11 @@
 from copy import deepcopy
+from typing import Dict
 
 from keras import backend as K
 from overrides import overrides
 
 from ..masked_layer import MaskedLayer
-from ...common.params import Params
+from ...common.params import pop_choice_with_default
```

Comment:
You would have to check when reading the code or when writing the code?  I don't think it's particularly obvious that `pop_choice_with_default("type", choices)` will return the first choice if "type" is not a key in the dict - you have to look at the docstring or know the method to know that.  And if that's true, is it much of a stretch to go from there to `pop_choice("type", choices, allow_default=True)`?  One of the first names I thought of for `allow_default` was `use_first_choice_as_default`, which is a lot more clear, but quite long.  What do you think?